packages/echo-base/src/errors/toError.ts

```diff
@@ -0,0 +1,68 @@
+import { BaseError } from './BaseError';
+
+export class NotAProperErrorObject extends BaseError {}
+
+/**
+ * Helper method: Useful with Try Catch(error), where error is of type unknown.
+ * Returns same error if it's already an error (or if it extends error),
+ * otherwise it converts it to type NotAProperErrorObject (with all properties preserved) which extends baseError
+ * @param error The error to convert (if needed)
+ * @returns Error or a an object which extends Error.
+ */
+export function toError(error: Error | BaseError | unknown): Error | BaseError {
+    if (error instanceof BaseError || error instanceof Error) {
+        return error;
+    }
+
+    const errorType = typeof error;
+    if (
+        errorType === 'string' ||
+        errorType === 'number' ||
+        errorType === 'boolean' ||
+        errorType === 'bigint' ||
+        errorType === 'symbol'
+        // 'undefined' |
+        // 'function'
+    ) {
+        let message = 'unknown';
+        switch (errorType) {
+            case 'string':
+                message = error as string;
+                break;
+            case 'number':
+                message = (error as number).toString();
+                break;
+            case 'boolean':
+                message = (error as boolean).toString();
+                break;
+            case 'bigint':
+                message = (error as bigint).toString();
+                break;
+            case 'symbol':
+                message = (error as symbol).toString();
+                break;
+            default:
+                break;
+        }
+        return new NotAProperErrorObject({
+            message: message,
+            exception: { argumentType: errorType }
+        });
+    }
+
+    if (errorType === 'object') {
+        const errorAsObject = error as object;
+        const properties = { ...errorAsObject } as { [key: string]: unknown };
+
+        const maybeName = (properties['name'] as string) ?? undefined;
```

Comment:
Would it make sense to rename / simplify here a bit? Not sure why it's needed to fallback to `undefined` values.
As far as I know `??` returns with the right hand side value if the left hand side is `null` or `undefined`.

So maybe this could be refactored a bit to something like this:
`````
const errorName = (properties['name'] as string) ?? '';
const errorMessage = (properties['message'] as string) ?? '';
const message = (`${errorName} ${errorMessage}`).trim()
`````

Reply:
Good input. I still want to call it maybeName/message, since we are just guessing there might be a property with the name and/or message field. Maybe indicates it might be null/missing (taken from functional languages).